src/elements/components/editable-users.tsx

```diff
@@ -1,40 +1,39 @@
 import React from 'react';
 import { BsPersonFillAdd } from 'react-icons/bs';
 import { MdRemoveCircle } from 'react-icons/md';
-import { User, UserId } from '../../lib/schema';
-import { asArray } from '../../lib/util';
+import { useUsers } from '../../lib/hooks/use-users';
+import { UserId } from '../../lib/schema';
 import { Link } from './link';
 
 export interface EditableUsersProps {
-    author: readonly UserId[];
-    users: ReadonlyMap<UserId, User>;
+    users: readonly UserId[];
     readonly?: boolean;
-    onAdd?: () => void;
-    onChange?: (newAuthor: UserId, index: number) => void;
-    onRemove?: (index: number) => void;
+    onChange?: (users: UserId[]) => void;
 }
 
-export function EditableUsers({ author, users, onChange, onAdd, onRemove, readonly }: EditableUsersProps) {
+export function EditableUsers({ users, onChange, readonly }: EditableUsersProps) {
+    const { userData } = useUsers();
+    const [randomUser] = userData.keys();
+
     return (
         <div className="flex flex-row flex-wrap gap-2">
             <strong className="m-0 flex flex-row flex-wrap gap-2 text-lg text-accent-600 dark:text-accent-500">
-                {asArray(author).map((userId, userIndex) => {
-                    if (!readonly && onChange && onRemove) {
+                {users.map((userId, index) => {
+                    if (!readonly && onChange) {
                         return (
                             <div
                                 className="flex flex-row gap-2"
-                                key={`${userId}${userIndex}`}
+                                key={`${userId} ${index}`}
                             >
                                 <select
-                                    defaultValue=""
                                     value={userId}
                                     onChange={(event) => {
-                                        const content = event.target.value as UserId;
-                                        onChange(content, userIndex);
+                                        const newUsers = [...users];
+                                        newUsers[index] = event.target.value as UserId;
+                                        onChange(newUsers);
                                     }}
                                 >
-                                    <option value="">Select user</option>
```

Comment:
Why get rid of this? I did this specifically so that it forces you to select one before adding more

Reply:
1. It's an invalid user ID that could be selected anywhere. You could just set all users to "Select user".
2. It doesn't force anything.

Reply:
it should have been preventing the "add user" button from showing until you select a user out of the list

Reply:
Yeah, not a good system. Adding invalid data into the DB and then hoping that the user cleans it up isn't a winning strategy IMO. Never write invalid data.

Reply:
makes sense

Reply:
I guess preferably what it would do is not write the data until a valid user is selected. but tbh, it doesn't really matter. this is just an internal tool after all